### DISCUSSION TITLE
Upserts do not seem to work on partial index on postgres

### DISCUSSION THREAD

**Discussion by beacoding** [supabase/supabase#12565] [Questions] [answered]
Looks like we cannot specify an `onConflict` and `where` together when we set a partial index on our postgres db.

We end up getting a  `unique or exclusion constraint matching the ON CONFLICT` when a partial index is used and we specify the key to use it for.

The solution to fix this is to filter on the index_predicate, but the client does not allow us to do this
https://stackoverflow.com/questions/42022362/no-unique-or-exclusion-constraint-matching-the-on-conflict

Are there any plans to support upserts of partial indices?

**Comment by steve-chavez** [comment, marked as answer]
@beacoding Yes, for now this is not supported directly but you can use a function, use the partial index inside and call it through [RPC](https://supabase.com/docs/reference/javascript/rpc).

> Are there any plans to support upserts of partial indices?

It's a reasonable addition, just so happens I'm working on `upsert`. Let's track it on https://github.com/supabase/postgrest-js/issues/403.

**Reply by beacoding** [reply]
thanks @steve-chavez !
